core/client/header-metadata/src/lib.rs

```diff
@@ -22,7 +22,7 @@ use parking_lot::RwLock;
 use lru_cache::LruCache;
 
 /// Set to the expected max difference between `best` and `finalized` blocks at sync.
-const LRU_CACHE_SIZE: usize = 5_000;
+const LRU_CACHE_SIZE: usize = 10_000;
```

Comment:
This test: https://github.com/paritytech/substrate/pull/3776/files#diff-c0982d5a95c967fde94ecf0a10d342dbR2252
Seems to rely that the value is 5000.

CC @andresilva